lib/client.js

```diff
@@ -48,6 +48,15 @@ function Client (broker, conn, req) {
 
   this.connDetails = null
 
+  // queue packets received before client fires 'connect' event. Prevents memory leaks on 'connect' event
+  this.parser._queue = []
+
+  this.on('connected', () => {
+    while (this.parser._queue.length > 0) {
+      handle(this, this.parser._queue.shift(), this._nextBatch)
+    }
```

Comment:
Instead of calling `shift()` just use a `for(;;)` loop and then assign `this.parser._queue` to null.